discoverer/src/main/java/eu/seaclouds/platform/discoverer/api/DiscovererAPI.java

```diff
@@ -0,0 +1,312 @@
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package eu.seaclouds.platform.discoverer.api;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import eu.seaclouds.platform.discoverer.core.Discoverer;
+import eu.seaclouds.platform.discoverer.core.Offering;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Date;
+import java.util.Set;
+
+@Path("/")
+public class DiscovererAPI {
+
+    private final Discoverer discoverer;
+    private org.json.simple.parser.JSONParser JSONParser;
+    static Logger log = LoggerFactory.getLogger(DiscovererAPI.class);
+
+    public DiscovererAPI(Discoverer discoverer) {
+        this.discoverer = discoverer;
+        this.JSONParser = new JSONParser();
+    }
+
+    @POST
+    @Path("/delete")
+    @Produces(MediaType.APPLICATION_JSON)
+    public DeleteRepresentation deleteOfferingById(@QueryParam("oid") String offerId)
+            throws IOException {
+
+        ArrayList<String> removedOfferingIds = new ArrayList<>();
+
+        /* if the offering id is valid and the discoverer is able to remove the offering associated
+         * then the offering id is added to the list of removed offering  */
+        if(this.discoverer.removeOffering(offerId)) {
+            removedOfferingIds.add(offerId);
+        }
+
+        return new DeleteRepresentation(removedOfferingIds);
+    }
+
+    @GET
+    @Path("/fetch_all")
+    @Produces(MediaType.APPLICATION_JSON)
+    public FetchAllRepresentation getAllOfferings() throws IOException {
+
+        /* collecting all the ids within the repository */
+        ArrayList<String> ids = new ArrayList<String>();
+        Collection<String> offeringIds = discoverer.getAllOfferingIds();
+        for (String offeringId : offeringIds) {
+            ids.add(offeringId);
+        }
+
+        Offering offering = this.discoverer.fetchOffer("all");
+        String offeringTOSCA = "";
+
+        if(offering != null) {
+            offeringTOSCA = offering.toscaString;
+        }
+
+        if (offeringTOSCA.isEmpty()) {
+            ids.clear();
+        }
+
+        return new FetchAllRepresentation(ids, offeringTOSCA);
```

Comment:
Unfortunately this does not apply here because this method is used to get all the offerings as a unique TOSCA YAML file, but it is fine for "fetchif" which precisely returns a list of Offerings.